app/Console/Commands/ExternalServices/ManageDiscord.php

```diff
@@ -83,13 +87,17 @@ protected function assignNickname()
         $this->discord->setNickname($this->account, $this->account->name);
     }
 
-    protected function grantRoles()
+    public function grantRoles()
     {
         $account = $this->account;
         $discord = $this->discord;
 
         $currentRoles = $discord->getUserRoles($account);
 
+        if ($account->isBanned && ! $currentRoles->contains($this->suspendedRoleId)) {
```

Comment:
Could we somehow, within this block, also remove any other roles that they have in $currentRoles? Regardless of what DiscordRole says they should have?

Perhaps just by extracting that logic (of giving the suspended role and removing all others) to a `$this->processSuspendedMember()` method and not letting them get to the `grantRoles()` or `removeRoles()` methods?